test-result-summary-client/src/Dashboard/Widgets/Graph/AcemairNodejs.jsx

```diff
@@ -0,0 +1,206 @@
+                const jdkDate = t.jdkDate; 
+                if ( !t.validAggregateInfo || element.metrics[0].name !== "Throughput" ) {
+                    return;
+                }
+                if(!resultsByJDKBuild[t.buildName]) {
+                    resultsByJDKBuild[t.buildName] = {};
+                }
+                resultsByJDKBuild[t.buildName][jdkDate] = resultsByJDKBuild[jdkDate] || [];
+                resultsByJDKBuild[t.buildName][jdkDate].push( {
+                    globalThroughput: element.metrics[0].value["mean"],
+                    additionalData: {
+                        mean: element.metrics[0].value["mean"],
+                        max: element.metrics[0].value["max"],
+                        min: element.metrics[0].value["min"],
+                        median: element.metrics[0].value["median"],
+                        stddev: element.metrics[0].value["stddev"],
+                        CI: element.metrics[0].value["CI"],
+                        validIterations: element.metrics[0].value["validIterations"],
+                        testId: (Array.isArray(t.tests) && t.tests.length > 0) ? t.tests[0]._id : null,
+                        hasChildren: t.hasChildren,
+                        parentId:t._id,
+                        buildName: t.buildName,
+                        buildNum: t.buildNum,
+                        nodeVersion,
+                        jdkDate
+                    },
+                } );
+            }
+        } );
+        Object.keys( resultsByJDKBuild ).forEach(( k, i ) => {
+            if(!globalThroughputs[k]) {
+                globalThroughputs[k] = [];
+            }
+            math.sort(Object.keys(resultsByJDKBuild[k])).forEach((a, b) => {
+                const date = a.substring(0,4) + "-" + a.substring(4,6) + "-" + a.substring(6,8) //Example: 2019-03-29
+                globalThroughputs[k].push( [new Date(date).getTime(), math.mean( resultsByJDKBuild[k][a].map( x => x['globalThroughput'] ) / scale), resultsByJDKBuild[k][a].map( x => x['additionalData'] )] );
+            });
+        } );
+
+        const displaySeries = baseLine;
+        for ( let key in globalThroughputs ) {
+            displaySeries.push( {
+                visible: key,
+                name: key,
+                data: globalThroughputs[key],
+                keys: ['x', 'y', 'additionalData']
+            } );
+        }
+
+        this.setState( { displaySeries } );
+    }
+
+    formatter = function() {
+        const x = this.x;
+        if ( this.point.additionalData ) {
+            let buildLinks = '';
+            let i = this.series.data.indexOf(this.point);
+            this.point.additionalData.forEach(( xy, i ) => {
+                const { testId, parentId, buildName, buildNum, hasChildren } = xy;
+                let childBuildLinks = '' + ` <a href="/output/test?id=${testId}">${buildName} #${buildNum}</a>`;
```

Comment:
Please use `const` for the values that do not change. The same for the rest variables.